### PR TITLE
Feature dynamo adapter - issue 2

### PR DESCRIPTION
**Pull request by manush-nayaka**
-to implement the UserDynamoDBAdaptor, to write unit test cases by mocking the boto3 library

### CODE REVIEW

File: domain/model/user.py [hunk 1]
```diff
@@ -1,8 +1,18 @@
+import inspect
+
 class User:
     """template class to hold user details"""
```
Comment (harnit-bakshi) [inline domain/model/user.py:4]: Please remove this comment, does not add much value

File: domain/model/user.py [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
     def __init__(self, **kwargs):
         for key in kwargs.keys():
             setattr(self, key, kwargs[key])
 
+    def asdict(self):
+        user_dict = dict()
+        for name in dir(self):
+            value = getattr(self, name)
```
Comment (harnit-bakshi) [inline domain/model/user.py:12]: Can you resolve value only after your check passes, that way we do not waste extracting value

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -33,6 +33,6 @@ jobs:
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
     - name: Test with pytest
       run: |
-        pip install -r requirements_test.txt
+        pip install -r requirements-test.txt
```
Comment (harnit-bakshi) [inline .github/workflows/workflow.yml:36]: According to python naming convention I think requirements_test.txt looks good

File: tests/test_dynamodb_adapter.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import pytest
+from unittest.mock import Mock, patch
+from infrastructure.adapter_persistence_dynamodb.user_dynamodb_adapter\
+    import UserDynamoDBAdapter
+from domain.model.user import User
+
+
+@pytest.fixture
```
Comment (harnit-bakshi) [inline tests/test_dynamodb_adapter.py:8]: Can we use @mock.patch('boto3') annotation

File: domain/model/user.py (hunk 1 above)
Comment (manush-nayaka) [reply domain/model/user.py]: fixed!!

File: domain/model/user.py (hunk 2 above)
Comment (manush-nayaka) [reply domain/model/user.py]: fixed!!

File: .github/workflows/workflow.yml (hunk 3 above)
Comment (manush-nayaka) [reply .github/workflows/workflow.yml]: fixed!!

File: tests/test_dynamodb_adapter.py (hunk 4 above)
Comment (manush-nayaka) [reply tests/test_dynamodb_adapter.py:8]: fixed!!

**Comment by harnit-bakshi** [comment]
LGTM

**Pull request merged**